GitUI/CommandsDialogs/FormDiff.cs

```diff
@@ -68,17 +68,18 @@ private FormDiff()
             _baseRevision = new GitRevision(baseId);
             _headRevision = new GitRevision(headId);
 
-            ObjectId? mergeBase;
-            if (_baseRevision.ObjectId.IsArtificial || _headRevision.ObjectId.IsArtificial)
+            ObjectId? baseMergeId = baseId.IsArtificial ? Module.GetCurrentCheckout() : baseId;
+            ObjectId? headMergeId = headId.IsArtificial ? Module.GetCurrentCheckout() : headId;
```

Comment:
I considered making the function lazy, but FormDiff is an edge case, the experience in Browse has more functionality and I just wanted to fix the issue.
(I have spent quite some time in optimizing less than this, but that is for more often executed code. It is generally only Git commands we need to optimize...)
If you make a release merge this, otherwise I might make this lazy.

Reply:
Changed the setup some to not try multi revision diff in FormDiff where it is not needed.
Refactored the multi-diff vs range-diff handling to improve readability